stock_available_to_promise_release_restrict_lot/models/stock_move.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2024 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+from odoo.osv.expression import AND
+
+
+class StockMove(models.Model):
+
+    _inherit = "stock.move"
+
+    def _previous_promised_qty_sql_lateral_where(self, warehouse):
+        sql, params = super()._previous_promised_qty_sql_lateral_where(warehouse)
+        sql += "AND m.restrict_lot_id is null"
+        return sql, params
+
+    def _get_previous_restricted_lot_moves_domain(self):
+        return [
+            "&",
+            ("restrict_lot_id", "=", self.restrict_lot_id.id),
+            "&",
+            ("state", "not in", ["done", "cancel"]),
+            "&",
+            ("id", "!=", self.id),
+            "|",
+            ("priority", ">", self.priority),
+            "|",
+            "&",
+            ("priority", "=", self.priority),
+            ("date_priority", "<", self.date_priority),
+            "&",
+            ("priority", "=", self.priority),
+            "&",
+            ("date_priority", "=", self.date_priority),
+            ("id", "<", self.id),
+        ]
```

Comment:
This domain is really hard to read. The use of the AND and OR operators from the expressions module should improve readability ant therefore maintenance.